packages/table/src/Row.tsx

```diff
@@ -0,0 +1,323 @@
+      let shouldDispatchHasNestedRows = false;
+      let shouldDispatchHasRowSpan = false;
+
+      React.Children.forEach(children, child => {
+        if (
+          isComponentType(child, 'Row') &&
+          !shouldDispatchHasNestedRows &&
+          !hasNestedRows
+        ) {
+          shouldDispatchHasNestedRows = true;
+        }
+
+        if (
+          isComponentType(child, 'Cell') &&
+          child.props.rowSpan > 1 &&
+          !hasRowSpan &&
+          !shouldDispatchHasRowSpan
+        ) {
+          shouldDispatchHasRowSpan = true;
+        }
+      });
+
+      if (shouldDispatchHasNestedRows) {
+        tableDispatch({
+          type: TableTypes.SetHasNestedRows,
+          payload: true,
+        });
+      }
+
+      if (shouldDispatchHasRowSpan) {
+        tableDispatch({
+          type: TableTypes.SetHasRowSpan,
+          payload: true,
+        });
+      }
+    }, [children, hasNestedRows, hasRowSpan, tableDispatch]);
+
+    // Iterating over children twice because generated memoized values have different dependants
+    const renderedChildren = React.useMemo(() => {
+      const Icon = isExpanded ? ChevronDownIcon : ChevronRightIcon;
+      const chevronButton = (
+        <IconButton
+          onClick={() => setIsExpanded(curr => !curr)}
+          aria-label="chevron"
+          className={iconButtonMargin}
+        >
+          <Icon aria-label="chevron" color={uiColors.gray.dark2} />
+        </IconButton>
+      );
+      const renderedChildren: Array<React.ReactElement> = [];
+      let hasSeenRow = false;
+
+      React.Children.forEach(children, (child, index) => {
+        if (isComponentType(child, 'Cell')) {
+          if (!child.props.children) {
+            return null;
+          }
+
+          renderedChildren.push(
+            React.cloneElement(child, {
+              children: (
+                <span className={truncation}>{child.props.children}</span>
+              ),
+              key: `${indexRef.current}-${index}`,
+              disabled: child.props.disabled || disabled,
+            }),
+          );
+        } else {
+          if (isComponentType(child, 'Row')) {
+            hasSeenRow = true;
+          }
+        }
+      });
+
+      if (hasSeenRow) {
+        renderedChildren[0] = React.cloneElement(renderedChildren[0], {
+          children: (
+            <>
+              {chevronButton}
+              <span className={truncation}>
+                {renderedChildren[0].props.children}
+              </span>
+            </>
+          ),
+          className: cx(displayFlex, className),
+          key: `${indexRef.current}-${renderedChildren[0].props.children}`,
+        });
+      }
+
+      return renderedChildren;
+    }, [children, disabled, className, isExpanded, setIsExpanded]);
+
+    // Iterating over children twice because generated memoized values have different dependants
+    const nestedRows = React.useMemo(() => {
+      let hasSeenFirstCell = false;
```

Comment:
What does this flag do? From my reading it seems un-necessary but I imagine I'm missing something.

Reply:
Ah this was from selectable logic. Can remove